snapshot_dbg_cli_tests/test_breakpoint_utils.py

```diff
@@ -283,3 +284,36 @@ def test_normalize_breakpoint_populates_id_if_provided(self):
 
     self.assertIn('id', obtained_bp)
     self.assertEqual('b-123', obtained_bp['id'])
+
+  def test_split_log_expressions_no_expressions(self):
+    self.assertEqual(split_log_expressions('Hi there.'), ('Hi there.', []))
+
+  def test_split_log_expressions_simple(self):
+    self.assertEqual(
+        split_log_expressions('a={a}, b={b}, c={c}'),
+        ('a=$0, b=$1, c=$2', ['a', 'b', 'c']))
+
+  def test_split_log_expressions_escaped_dollar(self):
+    self.assertEqual(
+        split_log_expressions('$ {abc$}$ $0'), ('$$ $0$$ $$0', ['abc$']))
+
+  def test_split_log_expressions_repeated_field(self):
+    self.assertEqual(
+        split_log_expressions('a={a}, b={b}, a={a}, c={c}, b={b}'),
+        ('a=$0, b=$1, a=$0, c=$2, b=$1', ['a', 'b', 'c']))
+
+  def test_split_log_expressions_nested_braces(self):
+    self.assertEqual(
+        split_log_expressions('a={{a} and {b}}, b={a{b{{cde}f}}g}'),
+        ('a=$0, b=$1', ['{a} and {b}', 'a{b{{cde}f}}g']))
```

Comment:
This actually highlights a peculiarity about the implementation.  iirc the contents of the braces don't really matter (they'll be treated as expressions by the agent) which means that braces do not need to be balanced after the first level.  The implementation seems more limiting than required.  That being said, it may be worth it to make the parsing simpler given the likely small number of impacted use cases.

Reply:
Per offline discussion we'll keep the implementation as is.